pkg/operator/resource/resourceapply/storage.go

```diff
@@ -48,3 +50,32 @@ func ApplyStorageClass(client storageclientv1.StorageClassesGetter, recorder eve
 	reportUpdateEvent(recorder, required, err)
 	return actual, true, err
 }
+
+// ApplyCSIDriverV1Beta1 merges objectmeta, does not worry about anything else
+func ApplyCSIDriverV1Beta1(client storageclientv1beta1.CSIDriversGetter, recorder events.Recorder, required *storagev1beta1.CSIDriver) (*storagev1beta1.CSIDriver, bool, error) {
+	existing, err := client.CSIDrivers().Get(required.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		actual, err := client.CSIDrivers().Create(required)
+		reportCreateEvent(recorder, required, err)
+		return actual, true, err
+	}
+	if err != nil {
+		return nil, false, err
+	}
+
+	modified := resourcemerge.BoolPtr(false)
+	existingCopy := existing.DeepCopy()
+
+	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
+	if !*modified {
+		return existingCopy, false, nil
+	}
+
+	if klog.V(4) {
+		klog.Infof("CSIDriver %q changes: %v", required.Name, JSONPatchNoError(existing, existingCopy))
+	}
+
+	actual, err := client.CSIDrivers().Update(existingCopy)
```

Comment:
Do CSI drivers have any immutable fields?  Those can sometimes cause issues.  You don't have to fix them all now, but you can look at what we did for secrets to handle the situation